src/Microsoft.Azure.WebJobs.Host/Indexers/MethodInfoExtensions.cs

```diff
@@ -16,7 +16,7 @@ public static string GetFullName(this MethodInfo methodInfo)
                 throw new ArgumentNullException("methodInfo");
             }
 
-            return String.Format(CultureInfo.InvariantCulture, "{0}.{1}", methodInfo.DeclaringType.FullName, methodInfo.Name);
+            return String.Format(CultureInfo.InvariantCulture, "{0}.{1}", methodInfo.DeclaringType.FullName, Utility.GetFunctionName(methodInfo));
```

Comment:
Correct - though I appreciate what @pougetat was trying to do here. There are several log statements written in FunctionIndexer that use this GetShortName method (e.g. [here](https://github.com/Azure/azure-webjobs-sdk/blob/f386e54e5000ddddf255f529689ab2a9c3ee78cd/src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs#L116) and [here](https://github.com/Azure/azure-webjobs-sdk/blob/f386e54e5000ddddf255f529689ab2a9c3ee78cd/src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs#L175)) where we need to log the name taking FunctionNameAttribute into account.

I agree we need to leave these extension methods as is, since we want to use the actual method info in some cases. Instead update all the log statements (e.g. the ones referenced above) to use Utility.GetFunctionName.

Note that [here](https://github.com/Azure/azure-webjobs-sdk/blob/f386e54e5000ddddf255f529689ab2a9c3ee78cd/src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs#L392) you can see how in the FunctionDescriptor we construct, we have a special LogName property that takes FunctionNameAttribute into account, and we use for logging. Other properties of the descriptor are derived directly from the MethodInfo.

Reply:
Quick question : should Utility.GetFunctionName use GetShortName instead of the methodInfo.name int the event that there is no name attribute ?

In doing this, an indexing error will still enable logging of 'class.method' in the event that FunctionNameAttribute is null instead of just 'method'.